server/shadow/shadow_client.c

```diff
@@ -938,7 +939,8 @@ void* shadow_client_thread(rdpShadowClient* client)
 	peer->update->SurfaceFrameAcknowledge = (pSurfaceFrameAcknowledge) shadow_client_surface_frame_acknowledge;
 
 	StopEvent = client->StopEvent;
-	UpdateEvent = subsystem->updateEvent;
+	UpdateSubscriber = shadow_multiclient_get_subscriber(subsystem->updateEvent);
```

Comment:
I notice FreeRDP project is cleaning code without return check recently. 
I agree that we should check NULL return, I need to think about how to define the behavior if we have NULL returned. So do the other one.